### PR TITLE
fix: Item Tax Template by Company Permissions

### PR DESCRIPTION
**Pull request by ahmadpak**
Currently if multiple companies are being operated and Item Tax Template is created for each one. The user with a specific company permission is still able to see all the Item Tax Templates. This fix ensures that the appropriate Item Tax templates are applied

image
image
image

### CODE REVIEW

**Review by deepeshgarg007 — CHANGES_REQUESTED** [review]
@ahmadpak 
- A patch is needed to update company in the Item Tax Template to update existing records
- Also there is no need to add separate column for company in "Item Tax" table since its fetched from Item Tax Template, can be fetched from Item  Tax Template itself in `_get_item_tax_template`

**Comment by ahmadpak** [comment]
Ok.

**Comment by ahmadpak** [comment]
@deepeshgarg007 Please review the changes. 

Regards

File: erpnext/patches.txt [hunk 1]
```diff
@@ -706,3 +706,6 @@ execute:frappe.delete_doc_if_exists("DocType", "Bank Reconciliation")
 erpnext.patches.v13_0.move_doctype_reports_and_notification_from_hr_to_payroll #22-06-2020
 erpnext.patches.v13_0.move_payroll_setting_separately_from_hr_settings #22-06-2020
 erpnext.patches.v13_0.check_is_income_tax_component #22-06-2020
+execute:frappe.reload_doc("HR", "doctype", "Employee Advance")
```
Comment (deepeshgarg007) [inline erpnext/patches.txt]: Remove this two unrelated patches, already there above in patches.txt file

File: erpnext/patches/v12_0/update_item_tax_template_company.py [hunk 2]
```diff
@@ -0,0 +1,11 @@
+from __future__ import unicode_literals
+import frappe
+
+def execute():
+    item_tax_template_list = frappe.get_list('Item Tax Template')
```
Comment (deepeshgarg007) [inline erpnext/patches/v12_0/update_item_tax_template_company.py:5]: Reload the `Item Tax Template` doc before running the patch, companies are not getting saved properly in all the templates on running the patch first time, I had to rerun it again manually. Rest all seems fine

**Review by deepeshgarg007 — CHANGES_REQUESTED** [review]

File: erpnext/patches/v12_0/update_item_tax_template_company.py (hunk 2 above)
Comment (ahmadpak) [reply erpnext/patches/v12_0/update_item_tax_template_company.py:5]: Ok thanks.

**Comment by deepeshgarg007** [comment]
@ahmadpak any updates?

**Comment by ahmadpak** [comment]
@deepeshgarg007 I will push the changes today. Sorry I have been very busy in past few days.

**Pull request closed without merging**